### PR TITLE
[libigl] Fix configure error and dependencies

### LINKED ISSUE #8811: [libigl] build failure
**Host Environment**
- OS: Windows
- Compiler: revision

**To Reproduce**
`.\vcpkg.exe install libigl[imgui,embree,opengl,glfw,png] --recurse`

**Failure logs** 
```
Building package libigl[core,embree,glfw,imgui,opengl,png]:x64-windows...
-- Using cached C:/Libraries/vcpkg/downloads/libigl-libigl-f6b406427400ed7ddb56cfc2577b6af571827c8c.tar.gz
-- Using source at C:/Libraries/vcpkg/buildtrees/libigl/src/f571827c8c-b1a8caf56b
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: ninja -v
    Working Directory: C:/Libraries/vcpkg/buildtrees/libigl/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Libraries\vcpkg\buildtrees\libigl\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:296 (vcpkg_execute_required_process)
  ports/libigl/portfile.cmake:46 (vcpkg_configure_cmake)
  scripts/ports.cmake:94 (include)


Error: Building package libigl:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libigl:x64-windows
  Vcpkg version: 2019.09.12-nohash
```

**Additional context**
This related to issue: https://github.com/microsoft/vcpkg/issues/8809
After removing nanovg, and install libigl, I got this build failure.

### LINKED ISSUE #12442: [libigl] build failure with feature <imgui>
**Host Environment**
- OS: [Windows 10 Version 10.0.19041.388]
- Compiler: Visual Studio 2019 Version 16.6.4 Tools 14.26.28801

**To Reproduce**
`./vcpkg install libigl[core,glfw,imgui,png,xml]:x64-windows`

**Failure logs** 
- Cut and past the appropriate build messages from the console output.
`Computing installation plan...`
**`Package imgui does not have a bindings feature`**
- Please attach any additional failure logs mentioned in the console output.
None

**Additional context**
It seems port imgui had feature `bindings` changed to multiple separate bindings (commit https://github.com/microsoft/vcpkg/commit/0b78ddc14bf513ddc33fe70099c5993d789c7081) , making the imgui feature `bindings` invalid.

### PR DESCRIPTION
**Pull request by JackBoosY**
This PR will:
- Fix configure error
- Remove feature png due to source issue [#1199](https://github.com/libigl/libigl/issues/1199)
- Remove feature python due to python binding are in the process of being redone.
- Remove feature test
- Feature imgui is enabled only during static builds
- Since no symbol was exported, set it to only support static builds.

Fixes #8811 #12442

All features are tested pass in `x86-windows`, `x64-windows` and `x64-windows-static`.

### CODE REVIEW

**Comment by L-JIN** [comment]
I couldn't find imgui feature in libigl using ```#include <igl/opengl/glfw/View.h>``` with ```libigl[core, opengl, glfw, imgui]:x64-windows``` installed, so I have merged this PR locally for a fast try. But upgrading vcpkg and auto-rebuilding didn't solve the problem. 

You mentioned "Feature imgui is enabled only during static builds", but I still get ```BUILD_FAILED``` when installing static version.
```
$ .\vcpkg.exe install libigl[imgui]:x64-windows-static
Computing installation plan...
The following packages will be built and installed:
    libigl[core,imgui]:x64-windows-static
Starting package 1/1: libigl:x64-windows-static
Building package libigl[core,imgui]:x64-windows-static...
-- Using cached C:/Users/Jin Li/vcpkg/downloads/libigl-libigl-f6b406427400ed7ddb56cfc2577b6af571827c8c.tar.gz
-- Using source at C:/Users/Jin Li/vcpkg/buildtrees/libigl/src/f571827c8c-c8ef868db9
CMake Error at ports/libigl/portfile.cmake:39 (message):
  Feature imgui does not support static build currentlly
Call Stack (most recent call first):
  scripts/ports.cmake:90 (include)
```

**Comment by JackBoosY** [comment]
@L-JIN That's why this PR marked work in progress now.

**Comment by L-JIN** [comment]
@JackBoosY Sorry, I thought this is all finished and waiting to be merged.

**Comment by JackBoosY** [comment]
@L-JIN Feature `imgui` only support dynamic build.

**Review by PhoebeHui — APPROVED** [review]

**Comment by L-JIN** [comment]
@JackBoosY Glad to know that. Thank you!

**Comment by JackBoosY** [comment]
The test `microsoft.vcpkg.pr (Run PR Test x64_osx) ` is deprecated.

File: ports/libigl/CONTROL [hunk 1]
```diff
@@ -1,8 +1,9 @@
 Source: libigl
-Version: 2.1.0-2
+Version: 2.2.0-2
```
Comment (strega-nil) [inline ports/libigl/CONTROL:2]:
```suggestion
Version: 2.2.0
```

**Review by strega-nil — CHANGES_REQUESTED** [review]
I think the port version should be rev'd back since there's a new version, right?

**Comment by JackBoosY** [comment]
@strega-nil We can do it step by step.

**Pull request merged**